File: server/Models/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Backend.Models
{
    public class Order
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public int UserId { get; set; }
      
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalPrice { get; set; }

        public DateTime OrderDate { get; set; }

        public DateTime DeliveryDate { get; set; }

        public string? Note { get; set; }

        public string? Status { get; set; }
        public User User { get; set; } = null!;
       
        [JsonIgnore]
        public ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();

    }
}

File: mobile/src/types/order.ts
import { BaseModel } from "../base/model";
import { Product } from "./product";

export interface Order extends BaseModel {
  userId: number;
  totalPrice: number;
  orderDate: string;
  deliveryDate: string;
  note: string;
  status: string;

  orderDetails: OrderDetails[];
}

interface OrderDetails {
  orderId: number;
  productId: number;
  quantity: number;
  price: number;
  product: Product;
}
